### PR TITLE
Feature/add feature for returning correlation results

### PR DESCRIPTION
**Pull request by ElRaptorus**
## What did you change?

- When starting a process, it is now determined whether or not the process is actually a subprocess (based on the given payload, which can now - optionally - contain a callerId)
- Add functions for retrieving a given correlations processInstanceResults
- Remove the "correlation" cache crutch and instead use the `ProcessEntity` property correlationId, which was recently introduced to the process engine packages

## How can others test the changes?

There are integrationtests for this in the meta repo. It's easiest to run them and look at the results.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: ioc_module.js [hunk 1]
```diff
@@ -2,30 +2,30 @@
...
 const {
   ConsumerApiIamService,
-  ConsumerApiService,
   ConsumerApiProcessEngineAdapter,
+  ConsumerApiService,
 } = require('./dist/commonjs/index');
 
 function registerInContainer(container) {
+
+  // TODO: Temporary workaround until the IdentityServer is in place.
```
Comment (cmg-dev) [inline ioc_module.js]: Bitte die IdentityServer referenz löschen. Und das ist kein TODO.

So kann man das schreiben:

`// Temporary workaround until a suitable authority (identity provider) is in place.`

File: src/consumer_api.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {
   EventList,
   EventTriggerPayload,
   IConsumerApiService,
+  ICorrelationResult,
```
Comment (cmg-dev) [inline src/consumer_api.ts]: Ich bin noch nicht so richtig fan von dem ICorrelationResult.

Da denke ich noch einmal drüber nach.

Comment (ElRaptorus) [reply src/consumer_api.ts]: Hab da auch nochmal drüber nachgedacht.
Die Typisierung als solche will ich gerne bei behalten. EInfach damit man eine konkrete Bezeichnung in der Hand hat, was die Funktion zurück gibt.
Aber ich wär auch für eine generischere Implementierung, damit das Prozessergebnis ungekapselt zurückgegeben werden kann.

Wie wärs mit sowas hier: 

```TypeScript
export interface ICorrelationResult {
  [field: string | object]: any;
}
```

**Comment by ElRaptorus** [comment]
@cmg-dev done, siehe contracts

**Pull request merged**